fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -1798,14 +1798,43 @@ Version TagPartitionedLogSystem::getPeekEnd() const {
 		return std::numeric_limits<Version>::max();
 }
 
+std::vector<Reference<LocalitySet>> TagPartitionedLogSystem::getPushLocationsForTags(
+    std::vector<int>& fromLocations) const {
+	std::vector<Reference<LocalitySet>> restrictedLogSets;
+	for (auto& log : tLogs) {
+		int locationOffset = 0;
+
+		if (!log->isLocal || !log->logServers.size()) {
+			locationOffset += log->logServers.size();
+			continue;
+		}
+		std::vector<LocalityEntry> e;
+		for (int i : fromLocations) {
+			if (i >= locationOffset && i < locationOffset + log->logServers.size()) {
+				e.emplace_back(LocalityEntry(i));
+			}
```

Comment:
It appears you are making certain assumptions about the range of "i" (based on the If-condition above). Probably you should add an assert that validates those assumptions, thanks!

Reply:
Here, the code loops over all local tLog sets, and checks if any entries within 'fromLocations' is in one of those sets. 'i' just tracks which one of the positions is currently being checked.